src/site.rs

```diff
@@ -30,8 +33,16 @@ impl Site {
         self.options.is_valid()
     }
 
+    pub fn get_site_name(&self) -> String {
+        if self.options.filename.is_none() {
+            filename::basename(&self.url)
+        } else {
+            self.options.filename.clone().unwrap()
+        }
```

Comment:
`if let Some(filename)` would be way better than this `is_none else unwrap` magic. Using `unwrap_or_else` is even simpler in my opinion.

```suggestion
        self.options
            .filename
            .clone()
            .unwrap_or_else(|| filename::basename(&self.url))
```